src/container.ts

```diff
@@ -36,15 +37,17 @@ export type Configuration = {
 export class Container {
     private readonly configuration: Configuration;
 
-    private context: Context;
+    private context?: Context;
 
-    private tracker: Tracker;
+    private tracker?: Tracker;
 
-    private evaluator: Evaluator;
+    private evaluator?: Evaluator;
 
-    private beaconChannel: OutputChannel<Beacon>;
+    private cidAssigner?: CidAssigner;
 
-    private beaconQueue: MonitoredQueue<string>;
+    private beaconChannel?: OutputChannel<Beacon>;
+
+    private beaconQueue?: MonitoredQueue<string>;
```

Comment:
Why make this properties optional?

Reply:
It's lazily initialized and was wrongly typed